waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption.py

```diff
@@ -0,0 +1,38 @@
+from threading import Thread
+from time import sleep
+
+
+class ConnectionInterruption(Thread):
+    """
+    A scheduler to drop incoming traffic to a port periodically.
+    """
+
+    def __init__(self, service, interruption_length, no_of_interruptions, service_node_index):
+        """
+            Construct a new 'ConnectionInterruption' object.
+
+            :param service: The service having incoming traffic to a port blocked
+            :param interruption_length: The interval(milliseconds) between two interruptions and the length of each interruption
+            :param no_of_interruptions: Total number of network connection interruptions
+            :param service_node_index: A service node used for this test
+        """
+        Thread.__init__(self)
+        self.service = service
+        self.interruption_length = interruption_length
+        self.interruptions = no_of_interruptions
+        self.service_node_index = service_node_index
+
+    def run(self):
+        node = self.service.nodes[self.service_node_index]
+        for interruption in range(self.interruptions):
+            # let waltz server receive transactions as usual
+            sleep(self.interruption_length/1000)
```

Comment:
It seems like we are accepting `interruption_length` as milliseconds, but we are dropping the relevant precision and are converting it to seconds. Any specific reason why we are setting an expectation that the time is in ms. Can we change the param to accept seconds instead?

Reply:
I believe that python doesn't convert the number to an integer like Java, but a decimal point number instead. But I can definitely change interruption_length to be represented in seconds. Interval between transactions is represented in milliseconds, on the other hand timeout is in seconds and I feel like there won't be a need to set _interruption_length_ to let's say 1.5s and 1s or 2s seconds will do as well.